src/main/java/core/basesyntax/bookstore/security/ExceptionHandlerFilter.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.bookstore.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import core.basesyntax.bookstore.dto.exception.ExceptionDto;
+import io.jsonwebtoken.JwtException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class ExceptionHandlerFilter extends OncePerRequestFilter {
```

Comment:
Why do you need it? can you use regular GlobalExceptionHandler?

Reply:
Can't - if request runs into exception in filter - it doesn't reach controller, therefore controller advice (which is GlobalExceptionHandler) doesn't react